Python/pystate.c

```diff
@@ -286,6 +286,9 @@ PyInterpreterState_Clear(PyInterpreterState *interp)
     }
     HEAD_UNLOCK(runtime);
 
+    PyDict_Clear(interp->sysdict);
+    PyDict_Clear(interp->builtins);
```

Comment:
I would prefer to only clear them at the end of the function, and explain that they should only be cleared at the end, since clearing other attributes can execute arbitrary Python code which requires sysdict and builtins.

Reply:
Thanks, updated.